terraform/modules/cos/main.tf

```diff
@@ -13,13 +15,13 @@ module "grafana" {
 }
 
 module "loki" {
-  source     = "git::https://github.com/canonical/observability//terraform/modules/loki"
+  source     = "git::https://github.com/canonical/observability//terraform/modules/loki?ref=self-monitoring"
```

Comment:
Drop the ref?
BTW should our modules have an input variable for the `ref`? (Separate issue? Only if you think it's a good idead)